src/main/java/org/folio/consortia/service/impl/SharingRoleCapabilityService.java

```diff
@@ -85,9 +101,51 @@ protected void validateSharingConfigRequestOrThrow(UUID roleId,
     }
   }
 
+  @Override
+  protected void syncConfigWithTenants(Set<String> sharedConfigTenants, SharingRoleCapabilityRequest request) {
+    String roleName = request.getRoleName();
+    String centralTenantId = folioExecutionContext.getTenantId();
+    log.debug("syncConfigWithTenant:: Trying to syncing sharing role table with role capabilities for role '{}' in central tenant '{}'",
+      request.getRoleName(), centralTenantId);
+
+    if (sharingRoleRepository.existsByRoleNameAndTenantIdAndIsCapabilitiesSharedTrue(roleName, centralTenantId)) {
+      log.info("syncConfigWithTenant:: Role '{}' and capabilities with tenant '{}' already exists, Syncing with other tenants: {}",
+        request.getRoleName(), centralTenantId, sharedConfigTenants);
+
+      sharedConfigTenants.stream()
+        .filter(tenantId -> !tenantId.equals(centralTenantId))
+        .forEach(memberTenantId -> syncSharingRoleWithRoleCapabilitiesInTenant(roleName, memberTenantId));
+      return;
+    }
+
+    syncSharingRoleWithRoleCapabilitiesInTenant(roleName, centralTenantId);
+  }
+
+  private void syncSharingRoleWithRoleCapabilitiesInTenant(String roleName, String tenantId) {
+    try (var ignored = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, moduleMetadata, folioExecutionContext))) {
```

Comment:
we invoke any sharing-related functionality with system user context

Reply:
used system user scope, thanks to inform